src/core/module.rs

```diff
@@ -307,3 +307,14 @@ impl<T: AsInnerInstance> AsInstance for T {
         Ok(())
     }
 }
+
+trait TryAsString {
+    fn try_as_string(&self) -> Result<String, std::str::Utf8Error>;
+}
+
+impl TryAsString for ffi::WasmEdge_String {
+    fn try_as_string(&self) -> Result<String, std::str::Utf8Error> {
+        let cstr = unsafe { std::ffi::CStr::from_ptr(self.Buf as *const _) };
+        Ok(cstr.to_str()?.to_string())
+    }
+}
```

Comment:
Can use std traits in Rust to replace it.

Reply:
No, in this crate `ffi::WasmEdge_String` is a foreign type belonging to a different crate, and any std trait would be a foreign  trait.
We can't define foreign traits on foreign types, at least one of them has to be defined locally.

A trait from std could be used if we were in the `wasmedge-sys` crate.

Reply:
ok, I will impl std trait in wasmedge-sys.

Reply:
I think the trait should be [`TryInto`](https://doc.rust-lang.org/std/convert/trait.TryInto.html) (as opposed to `Into`  that's currently implemented in `wasmedge_sys_ffi`)